packages/fxa-customs-server/lib/sentry.js

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const Hoek = require('@hapi/hoek');
+const Sentry = require('@sentry/node');
+const version = require('../package.json').version;
+
+async function configureSentry(server, config, log) {
+  const sentryDsn = config.sentryDsn;
+  if (sentryDsn) {
+    Sentry.init({
+      dsn: sentryDsn,
+      release: version,
+      integrations: [
+        new Sentry.Integrations.LinkedErrors({ key: 'jse_cause' }),
+      ],
+    });
+
+    Sentry.configureScope(scope => {
+      scope.setTag('process', 'customs_server');
+    });
+
+    // Attach a new Sentry scope to the request for breadcrumbs/tags/extras
+    server.ext({
+      type: 'onRequest',
+      method(request, h) {
+        request.sentryScope = new Sentry.Scope();
+        return h.continue;
+      },
+    });
+    log.info({ op: 'sentryEnabled' });
+  } else {
+    log.info({ op: 'sentryDisabled' });
+  }
+
+  // Sentry handler for hapi errors
+  server.events.on({ name: 'request', channels: 'error' }, (request, event) => {
+    const err = (event && event.error) || null;
+
+    if (sentryDsn) {
+      let exception = '';
+      if (err && err.stack) {
+        try {
+          exception = err.stack.split('\n')[0];
+        } catch (e) {
+          // ignore bad stack frames
+        }
+      }
+      Sentry.withScope(scope => {
+        scope.addEventProcessor(_sentryEvent => {
+          const sentryEvent = Sentry.Handlers.parseRequest(
```

Comment:
This was the primary thing I had meant should have been included, didn't mean you had to copy the whole thing over! Sorry.